test/extended/networking/internal_ports.go

```diff
@@ -147,6 +153,24 @@ var _ = ginkgo.Describe("[sig-network] Internal connectivity", func() {
 		}
 		errs := parallelTest(6, testFns)
 		o.Expect(errs).To(o.Equal([]error(nil)))
+		for _, pod := range pods.Items {
+			o.Eventually(func() float64 {
```

Comment:
`Eventually should be < 0.01` is not the same thing as `must be < 0.01`.  Why are we retrying?

Reply:
As I mentioned in the description to this PR. We'll have spurious flakes in CI saying that UDP tests failed whereas we just didn't wait for the traffic to reach its destination. It's now 1000 UDP packets / node, so it should be around 7-10.000 packets per test cluster.

> If we run the check too early: all messages might not have been delivered. I am thus using a Eventually comparison with a timeout of 10 seconds. This proved, on my machine to be sufficient to get a 100 % rate of 10.000 UDP packets passing.

Reply:
We want the total flake rate to be below a percentage.  Eventually could have 9 runs that have 99% failure rate and then 1 run that has 1% rate and eventually would pass, but we don't want that.  We want to run up to 10 times, and if failure rate across all runs is below 1%, pass.

Reply:
I don't follow. Are your referring to a CI run? This is not testing anything related to that. It's strictly testing "for every test pod on the cluster what % of UDP packets did it drop, w.r.t how many it was supposed to receive", and `Eventually` is needed to properly evaluate that, as to ensure that all packets that might have reached the pod, do so. If any pods saw a UDP drop rate below 1 % during CI run X, after having waited for a maximum of 10 seconds for all packets to arrive, then we fail instantly the test. This will then allow us to compare failed CI runs for this test and see how many are impacted by this condition.

Reply:
Ah, I see what you are trying to do now.  I don't like this structure because it's too easy for accidental failures - there are ways that you would fail with an empty TCP dump and give a false positive.  The grep for UDP failures should happen at the point of testing on line 145 above, as a second command execution.  If err = nil return, if type = tcp, return right away, if type = udp do the grep.

Reply:
I don't see what you mean. If the TCP dump is empty the value `expectedUDPPacketCount` will be 0 as a result of `strconv.Atoi(res)` converting a "non-parseble" integer value, which will mean we fail the test (which is also what we want to do, in such a case).  

Anyways, I have changed the test and done what you requested.